### ISSUE TITLE
Cloud storage support?

### ISSUE THREAD

**Issue by tillkruss** [tgeisse/UserDefaultsObservation#1]
Do you have any plans to support NSUbiquitousKeyValueStore in this too?

**Comment by tgeisse** [comment]
I will add it to my list and I can take a look! Thank you for the idea.

**Comment by tgeisse** [comment]
I have a couple of design questions for you. I can probably get this to work, but want your take as someone who might use this on what the best paradigm is.

1. The demo app from Apple's documentation uses UserDefaults to cache the value. Would you always expect this to be the case or would you want control over when to use UserDefaults and when to only use cloud storage without local caching?
2. While your input on the first question will largely dictate this, what do you see as the better way to mark a variable for cloud storage?

**Using the current macro for a property, UserDefaultsObservableProperty, and providing the cloud details**
```swift
@ObservableUserDefaultsProperty("userDefaultsKey", cloudStorage: .init(key: "cloudName", onNotificationAction: .action))
var olderKey = 50
```

**Or having a new macro that separates UserDefaultsObservableProperty and this one.**
```swift
@UbiquitousKeyValueStore(key: "username_key", onNotificationAction: .action)
var username: String? = nil
```

**Comment by tgeisse** [comment]
@tillkruss  - this is now available in version 0.5.0

**Comment by tgeisse** [comment]
@tillkruss  - this is now available in version 0.5.0

**Comment by tillkruss** [comment]
Sweet, I'll take a look next week!
